prereise/cli/data_sources/solar_data.py

```diff
@@ -0,0 +1,182 @@
+from datetime import datetime
+
+from powersimdata.input.grid import Grid
+
+from prereise.cli.constants import (
+    DATE_FMT,
+    END_DATE_HELP_STRING,
+    FILE_PATH_HELP_STRING,
+    REGION_CHOICES,
+    START_DATE_HELP_STRING,
+)
+from prereise.cli.data_sources.data_source import DataSource
+from prereise.cli.helpers import validate_date, validate_file_path, validate_year
+from prereise.gather.solardata.ga_wind import ga_wind
+from prereise.gather.solardata.nsrdb import naive, sam
+
+API_KEY_HELP_STRING = "API key that can requested at https://developer.nrel.gov/signup/"
+NAIVE_STRING = "naive"
+SAM_STRING = "sam"
+
+
+class SolarDataGriddedAtmospheric(DataSource):
+    @property
+    def command_name(self):
+        """See :py:func:`prereise.cli.data_sources.data_source.DataSource.command_name`
+
+        :return: (*str*)
+        """
+        return "solar_data_ga"
+
+    @property
+    def command_help(self):
+        """See :py:func:`prereise.cli.data_sources.data_source.DataSource.command_help`
+
+        :return: (*str*)
+        """
+        return "Download solar data from the Gridded Atmospheric Wind Integration National Dataset Toolkit"
+
+    @property
+    def extract_arguments(self):
+        """See :py:func:`prereise.cli.data_sources.data_source.DataSource.extract_arguments`
+
+        :return: (*str*)
+        """
+        return [
+            {
+                "command_flags": ["--region", "-r"],
+                "required": True,
+                "choices": REGION_CHOICES,
+                "type": str,
+                "action": "append",
+            },
+            {
+                "command_flags": ["--start_date", "-sd"],
+                "required": True,
+                "type": validate_date,
+                "help": START_DATE_HELP_STRING,
+            },
+            {
+                "command_flags": ["--end_date", "-ed"],
+                "required": True,
+                "type": validate_date,
+                "help": END_DATE_HELP_STRING,
+            },
+            {
+                "command_flags": ["--file_path", "-fp"],
+                "required": True,
+                "type": validate_file_path,
+                "help": FILE_PATH_HELP_STRING,
+            },
+            {
+                "command_flags": ["--key", "-k"],
+                "required": True,
+                "type": str,
+                "help": API_KEY_HELP_STRING,
+            },
+        ]
+
+    def extract(self, region, start_date, end_date, file_path, key, **kwargs):
```

Comment:
We should add a `grid_model` parameter that will be used to instantiate the `Grid` class. I f you look at the constructor of the `Grid class`, you will see that it takes a `source` parameter that defaults to `usa_tamu`. This is the only grid model we have right now but in the near future we will have a new one and we will need to fetch data for the wind/solar generators in it.

Reply:
Sounds good. Ideally the `Grid class` would have a property that has a list of the possible sources ( "usa_tamu", etc) that the code could use as choices for the parameter. That way we have it defined in one location and whenever a new source is added it's automatically picked up by the CLI tool

Reply:
should make a PR to add that first and then update this one? Or should I land this first, make that PR and come back and update this code?

Reply:
I don't have any preference. Whatever works best for you. Also do not hesitate to add a property in the `Grid` class.

Reply:
@rouille I'll update the `Grid` class in a separate PR and come back and update this code.